synapse/storage/databases/main/events_worker.py

```diff
@@ -1462,68 +1468,68 @@ async def have_seen_events(
             event_ids: events we are looking for
 
         Returns:
-            The set of events we have already seen.
+            The remaining set of events we haven't seen.
         """
+        set_tag(
+            SynapseTags.FUNC_ARG_PREFIX + "event_ids.length",
+            str(len(event_ids)),
+        )
 
         # @cachedList chomps lots of memory if you call it with a big list, so
         # we break it down. However, each batch requires its own index scan, so we make
         # the batches as big as possible.
 
-        results: Set[str] = set()
-        for chunk in batch_iter(event_ids, 500):
-            r = await self._have_seen_events_dict(
-                [(room_id, event_id) for event_id in chunk]
-            )
-            results.update(eid for ((_rid, eid), have_event) in r.items() if have_event)
+        remaining_event_ids: Set[str] = set()
+        for chunk in batch_iter(event_ids, 1000):
```

Comment:
Why the number `1000` specifically? If it's supposed to match with [DeferredCache.max_entries](https://github.com/matrix-org/synapse/blob/f9f03426de338ae1879e174f63adf698bbfc3a4b/synapse/util/caches/deferred_cache.py#L74), could they be made to refer to the same declaration somehow?

Reply:
It's just arbitrarily doubled. I don't know of a good heuristic to justify a better value. But I know we don't want the SQL queries to get too big which is why we're batching in the first place.

Reply:
> Good to know that we shouldn't exceed the max_entries of the cache though 

Oh, that was a very tentative "if"; mentioned it as a guess because it was mentioned in the PR discussion.